docs/product-business-resource-planner-guide/src/main/asciidoc/optimizer-NP-complete-vs-NP-hard-con.adoc

```diff
@@ -0,0 +1,26 @@
+[id='optimizer-NP-complete-vs-NP-hard-con']
+= A planning problem is NP-complete or NP-hard
+
+The provided use cases are _probably_ http://en.wikipedia.org/wiki/NP-complete[NP-complete] or harder.
+NP-complete means:
+
+* It is easy to verify a given solution to a problem in reasonable time.
+* There is no simple way to find the optimal solution of a problem in reasonable time.
+
+////
+[NOTE]
+====
+At least, none of the smartest computer scientists in the world have found such a silver bullet yet.
+But if they find one for 1 NP-complete problem, it will work for every NP-complete problem.
+
+In fact, there's a $ 1,000,000 reward for anyone that proves if http://en.wikipedia.org/wiki/P_%3D_NP_problem[such a silver bullet actually exists or not].
+
+====
+////
+
+The implication is that solving your problem is probably harder than you anticipated, because the two common techniques will not suffice:
+
+* A Brute Force algorithm (even a more advanced variant) will take too long.
+* A quick algorithm, for example in bin packing, _putting in the largest items first_ will return a solution that is far from optimal.
```

Comment:
This is a little confusing. I don't understand putting the largest items first.

Reply:
I'm including an external link to the Wikipedia entry for the Bin packing problem, as it's too complicated to explain in our doc and not really necessary. This should give it some context at least.